test/resource_users.test.js

```diff
@@ -220,6 +222,30 @@ describe('User Resource', function() {
       const badProfile = { profile: { bad: 'profile' } }
       return expect(users.set(badProfile)).to.eventually.be.rejectedWith(Error)
     })
+
+    it('should be able to receive transactionHash and onComplete callbacks', async () => {
+      let transactionHash, confCount, trReceipt
+
+      await users.set({
+        profile: {
+          firstName: 'Wonder',
+          lastName: 'Woman',
+        },
+        options: {
+          transactionHashCallback: (hash) => {
+            transactionHash = hash
+          },
+          confirmationCallback: (confirmationCount, transactionReceipt) => {
+            confCount = confirmationCount
+            trReceipt = transactionReceipt
+          }
+        }
+      })
+
+      expect(transactionHash).to.startsWith('0x')
+      expect(confCount).is.a('number')
```

Comment:
Isn't there a possible race condition here since it is not guaranteed the confirmationCallback will get called before this line is executed ?

One possible way to test callbacks I think is to put the test code within the callback itself. See for ex: https://gist.github.com/eswak/c54989f05c6f2ce4be00
Maybe something along the lines of
```
confirmationCallback: (confirmationCount, transactionReceipt) => {
  expect(confirmationCount).is.a('number')
  expect(trReceipt).is.a('object')
  if (confirmationCount == NUMBER_CONFIRMATIONS_TO_REPORT) {
    // this test is done, go to the next one
    done()
  }
```

Reply:
Ahh yes thanks @franckc . The approach with a done() callback is much better.